src/overlays/actors/ovl_En_Mt_tag/z_en_mt_tag.c

```diff
@@ -35,42 +40,474 @@ const ActorInit En_Mt_tag_InitVars = {
+    { -3933.0, 1000.0, 1479.0 },  { -3212.0, 1000.0, 1461.0 },  { -2805.0, 1000.0, 1645.0 },
+    { -2638.0, 1000.0, 2071.0 },  { -2823.0, 1000.0, 2422.0 },  { -3212.0, 1000.0, 2607.0 },
+    { -3785.0, 1000.0, 2977.0 },  { -4321.0, 1000.0, 3501.0 },  { -4654.0, 1000.0, 4185.0 },
+    { -4802.0, 1000.0, 4779.0 },  { -4672.0, 1000.0, 5426.0 },  { -4339.0, 1000.0, 6037.0 },
+    { -3748.0, 1000.0, 6314.0 },  { -2749.0, 1000.0, 6478.0 },  { -2453.0, 1000.0, 6922.0 },
+    { -2269.0, 1000.0, 7754.0 },  { -2453.0, 1000.0, 8309.0 },  { -3008.0, 1000.0, 8438.0 },
+    { -3304.0, 1000.0, 8179.0 },  { -3600.0, 1000.0, 7606.0 },  { -3600.0, 1000.0, 6885.0 },
+    { -3618.0, 1000.0, 4392.0 },  { -3600.0, 1000.0, 3855.0 },  { -3396.0, 1000.0, 3189.0 },
+    { -3396.0, 1000.0, 2283.0 },  { -3600.0, 1000.0, 818.0 },   { -3803.0, 1000.0, -88.0 },
+    { -4543.0, 1000.0, -2457.0 }, { -4543.0, 1000.0, -2938.0 }, { -4543.0, 1000.0, -3530.0 },
+    { -4284.0, 1000.0, -4333.0 }, { -3581.0, 1000.0, -4795.0 }, { -2805.0, 1000.0, -4850.0 },
+    { -1825.0, 1000.0, -4703.0 }, { -1326.0, 1000.0, -4166.0 }, { -1122.0, 1000.0, -3186.0 },
+    { -1085.0, 1000.0, -2059.0 }, { -1067.0, 1000.0, -912.0 },
+};
+
+/**
+ * Returns true if the specified position is in the finish line.
+ * The range extends a little bit beyond the finish line's in-game visual.
+ */
+s32 EnMttag_IsInFinishLine(Vec3f* pos) {
+    return Math3D_XZBoundCheck(-1261.0f, -901.0f, -1600.0f, -1520.0f, pos->x, pos->z);
+}
+
+/**
+ * Returns a value in PlayerCheatStatus that indicates if the player is cheating
+ * and, if so, what kind of cheating the player is performing.
+ */
+s32 EnMttag_CheckPlayerCheatStatus(Vec3f* pos) {
+    if (!(gSaveContext.eventInf[1] & 1)) {
+        if (Math3D_XZBoundCheck(-466.0f, -386.0f, -687.0f, 193.0f, pos->x, pos->z)) {
+            // The race hasn't started yet, but the player is beyond the starting line.
+            return CHEAT_FALSE_START;
+        }
+    } else if (Math3D_XZBoundCheck(-1127.0f, -1007.0f, -867.0f, -787.0f, pos->x, pos->z)) {
+        // The goal is actually quite close to the start, just behind a large wall.
+        // This checks if the player is in an area "behind" the goal that is not accessible
+        // in normal play; it can only be reached by climbing the wall somehow. Perhaps they
+        // were worried that players would find a way to climb the wall with a glitch, or
+        // perhaps they just wanted to punish people using cheat codes.
+        return CHEAT_TRYING_TO_REACH_GOAL_FROM_BEHIND;
+    }
+    return CHEAT_NO_CHEATING;
+}
+
+/**
+ * This function tries to find all four Race Gorons present in the racetrack.
+ * If it finds them, it stores a pointer to each one in the actor's struct.
+ * Returns true if all four Race Gorons are found.
+ */
+s32 EnMttag_AreFourRaceGoronsPresent(EnMttag* this, GlobalContext* globalCtx) {
+    Actor* actor = NULL;
+    s32 i = 0;
+    s32 ret;
```

Comment:
```suggestion
    s32 areAllGoronsFound;
```
Maybe this isn't the best name, alternatives might be `areGoronsPresent` or `goronsFound`. I think it can be a bit better than `ret` though